### ISSUE TITLE
Linux Binary Portability

### ISSUE THREAD

**Issue by bdwyertech** [aws/amazon-ssm-agent#244]
If I attempt to run a Amazon Linux built binary on another, non RHEL-like Linux distro (e.g. CoreOS, Ubuntu, etc.) it fails to execute.

```shell
-bash: bin/linux_amd64/amazon-ssm-agent: No such file or directory
```

A recent change to the Linux build process seems to be the culprit
https://github.com/aws/amazon-ssm-agent/blob/070cd85b2af5dd54a798bc60e9a0db4001527344/makefile#L136

It seems that buildmode=pie yields less portability for binaries as it forces external linking
https://github.com/golang/go/commit/eccd3ef526956ab404b8b4a724a9a3862c77f457

The same result can be achieved by copying or extracting the binaries from the latest SSM Agent RPM and attempting to run it on another platform.

**Comment by LiangChiAmzn** [comment]
Hello @bdwyertech,

Thank you for reaching out to us.

#### About CoreOS:

 SSM Agent currently does not support CoreOS yet. We will have the team to take a look at this.

#### About Ubuntu:

Please refer to the following documentation about installing SSM Agent on Ubuntu Server:

https://docs.aws.amazon.com/systems-manager/latest/userguide/sysman-manual-agent-install.html#agent-install-ubuntu-about-v16

In summary, you could either use [Snapcraft](https://snapcraft.io/amazon-ssm-agent) or download the `.deb` installer package to install SSM Agent.

Thank you!
